tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1173,40 +1173,10 @@ private[optimizer] abstract class OptimizerCore(
                 treeNotInlined
               }
             } else {
-              if (impls.forall(_.isForwarder)) {
-                val reference = impls.head
-                val areAllTheSame = getMethodBody(reference).body match {
-                  // Trait impl forwarder
-                  case ApplyStatic(ClassType(staticCls), Ident(methodName, _), _) =>
-                    impls.tail.forall(getMethodBody(_).body match {
-                      case ApplyStatic(ClassType(`staticCls`),
-                          Ident(`methodName`, _), _) => true
-                      case _ => false
-                    })
-
-                  // Bridge method
-                  case MaybeBox(Apply(This(), Ident(methodName, _), referenceArgs), boxID) =>
-                    impls.tail.forall(getMethodBody(_).body match {
-                      case MaybeBox(Apply(This(), Ident(`methodName`, _), implArgs), `boxID`) =>
-                        referenceArgs.zip(implArgs) forall {
-                          case (MaybeUnbox(_, unboxID1), MaybeUnbox(_, unboxID2)) =>
-                            unboxID1 == unboxID2
-                        }
-                      case _ => false
-                    })
-
-                  case body =>
-                    throw new AssertionError("Invalid forwarder shape: " + body)
-                }
-                if (!areAllTheSame) {
-                  // Not all doing the same thing
-                  treeNotInlined
-                } else {
-                  inline(allocationSites, Some(treceiver), targs, reference,
-                      isStat, usePreTransform)(cont)
-                }
+              if (canMultiInline(impls)) {
```

Comment:
Because the outer `if` distinguishes between 0, 1, and 2+ targets. Within the 2+ targets case, we test whether we can multi-inline them or not.